tutorials/persistance-comparison-series/PlayerPrefs/player_prefs.md

```diff
@@ -0,0 +1,436 @@
+        HitCount hitCount = JsonUtility.FromJson<HitCount>(jsonString);
+
+        // 7
+        if (hitCount != null)
+        {
+            hitCountUnmodified = hitCount.Unmodified;
+            hitCountShift = hitCount.Shift;
+            hitCountControl = hitCount.Control;
+        }
+    }
+}
+```
+
+The detection for the key that was pressed is identical to the extended example since it does not involve loading or saving any data but is just a check for the key during `Update()`:
+
+```cs
+private void Update() // 8
+{
+    // Check if a key was pressed.
+    if (Input.GetKey(KeyCode.LeftShift)) // 9
+    {
+        // Set the LeftShift key.
+        keyPressed = KeyCode.LeftShift; // 10
+    }
+    else if (Input.GetKey(KeyCode.LeftControl)) // 9
+    {
+        // Set the LeftControl key.
+        keyPressed = KeyCode.LeftControl; // 10
+    }
+    else // 11
+    {
+        // In any other case reset to default and consider it unmodified.
+        keyPressed = default; // 12
+    }
+}
+```
+
+In a very similar fashion `OnMouseDown()` needs to save the data whenever it's changed.
+
+```cs
+private void OnMouseDown()
+{
+    if (keyPressed == KeyCode.LeftShift) // 13
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountShift++; // 14
+    }
+    else if (keyPressed == KeyCode.LeftControl) // 13
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountControl++; // 14
+    }
+    else // 15
+    {
+        // Increment the hit count and set it to PlayerPrefs.
+        hitCountUnmodified++; // 16
+    }
+
+    // 17
+    HitCount hitCount = new();
+    hitCount.Unmodified = hitCountUnmodified;
+    hitCount.Shift = hitCountShift;
+    hitCount.Control = hitCountControl;
+
+    // 18
+    string jsonString = JsonUtility.ToJson(hitCount);
+    PlayerPrefs.SetString(HitCountKey, jsonString);
+    PlayerPrefs.Save();
+}
+```
+
+Compared to before you see that checking the key and increasing the counter (13 - 16) is basically unchanged except for the save part that is now a bit different.
+
+First, we need to create a new `HitCount` object (17) and assign the three counts. Using `JsonUtility.ToJson()` we can then (18) create a JSON string from this object and set it using the `PlayerPrefs`.
+
+Remember to also call `Save()` here to make sure data cannot get lost in case the game crashes without being able to call `OnApplicationQuit()`.
+
+Run the game and after you clicked the capsule a couple of time with or without Shift and Control, have a look at the result. The following screenshot shows the Windows registry which is where the `PlayerPrefs` get saved.
+
+The location is `HKEY_CURRENT_USER\SOFTWARE\Unity\UnityEditor\MongoDB Inc.\UnityPersistenceExample` and as you can see our JSON is right there, saved in plain text. Which is also one of the big down sides to keep in mind when using `PlayerPrefs`: data is not save and can easily be edited.
```

Comment:
That location is probably not the same for all users, so probably worth replacing `MongoDB Inc.` with `*your-company*` or something like that

Reply:
Also:
> data is not save

should probably be `data is not safe`?

Reply:
26fb8c0